extra/raylib/live-coding/live-coding.factor

```diff
@@ -0,0 +1,121 @@
+! Copyright (C) 2024 Dmitry Matveyev.
+! See https://factorcode.org/license.txt for BSD license.
+USING: kernel raylib.live-coding.glfw threads calendar
+namespaces vocabs.refresh continuations timers combinators
+accessors concurrency.messaging concurrency.conditions ;
+FROM: raylib => window-should-close is-key-pressed
+    get-window-handle ;
+IN: raylib.live-coding
+
+SYMBOL: lc-listener-thread
+SYMBOL: lc-game-thread
+SYMBOL: lc-sleep-duration
+
+!
+! Common words
+!
+
+<PRIVATE
+SYMBOL: lc-window-should-close?
+ERROR: lc-continue-or-not original ;
+
+: lc-enabled? ( -- ? ) lc-game-thread get ;
+
+: send-to-listener ( message -- ) lc-listener-thread get send ;
+: send-to-game ( message -- ) lc-game-thread get send ;
+
+: set-window-should-close ( -- ) t lc-window-should-close? set ;
+PRIVATE>
+
+!
+! On the game thread
+!
+
+<PRIVATE
+: window-should-close? ( -- ? )
+    window-should-close
+    lc-window-should-close? get or ;
+
+: with-listener-context ( quot -- )
+    f make-context-current
+    call
+    get-window-handle make-context-current ; inline
+
+: yield-to-listener ( -- )
+    [ lc-sleep-duration get sleep ] with-listener-context ;
+
+: with-rescue-to-listener ( quot -- )
```

Comment:
It's kind of hard to tell how this is nested. Here's how I would expect to see it:

```factor
: with-rescue-to-listener ( quot -- )
    lc-enabled? [
        [
            [
                \ lc-continue-or-not boa send-to-listener
                "Game: listening to receive" .
                receive .
            ] with-listener-context
        ] recover
    ] [ call ] if ; inline
```